modules/sdk-lib-mpc/src/tss/ecdsa-dkls/dkg.ts

```diff
@@ -1,22 +1,70 @@
 import { KeygenSession, Keyshare, Message } from '@silencelaboratories/dkls-wasm-ll-node';
-import { DeserializedBroadcastMessage, DeserializedMessages, DkgState } from './types';
+import { DeserializedBroadcastMessage, DeserializedMessages, DkgState, RetrofitData } from './types';
 import { decode } from 'cbor';
+import { encode } from 'cbor-x';
+import { bigIntToBufferBE } from '../../util';
+import { Secp256k1Curve } from '../../curves';
 
 export class Dkg {
   protected dkgSession: KeygenSession | undefined;
   protected dkgSessionBytes: Uint8Array;
   protected dkgKeyShare: Keyshare;
+  protected keyShareBuff: Buffer;
   protected n: number;
   protected t: number;
   protected chainCodeCommitment: Uint8Array | undefined;
   protected partyIdx: number;
   protected dkgState: DkgState = DkgState.Uninitialized;
+  protected dklsKeyShareRetrofitObject: Keyshare | undefined;
 
-  constructor(n: number, t: number, partyIdx: number) {
+  constructor(n: number, t: number, partyIdx: number, retrofitData?: RetrofitData) {
     this.n = n;
     this.t = t;
     this.partyIdx = partyIdx;
     this.chainCodeCommitment = undefined;
+    if (retrofitData) {
+      if (!retrofitData.xShare.y || !retrofitData.xShare.chaincode || !retrofitData.xShare.x) {
+        throw Error('xShare must have a public key, private share value, and a chaincode.');
+      }
+      if (n !== 3 || t !== 2) {
+        throw Error('only 2 of 3 retrofit is supported.');
+      }
+      if (retrofitData.bigSiList.length !== 2) {
+        throw Error("bigSiList should contain the other parties' Si's");
+      }
+      const big_s_list: Array<Array<number>> = [];
+      let j = 0;
+      for (let i = 0; i < n; i++) {
+        if (i === partyIdx) {
+          const secp256k1 = new Secp256k1Curve();
+          big_s_list.push(Array.from(bigIntToBufferBE(secp256k1.basePointMult(BigInt('0x' + retrofitData.xShare.x)))));
+        } else {
+          big_s_list.push(Array.from(Buffer.from(retrofitData.bigSiList[j], 'hex')));
+          j++;
+        }
+      }
+      const x_i_list = [new Array(32).fill(0), new Array(32).fill(0), new Array(32).fill(0)];
```

Comment:
I think we should move this whole retrofit related code out of the constructor and add jsdoc to explain the process.

Reply:
which part of the process you think needs to be in the jsdoc , this is a key refresh flow do you think we need a brief description of key refresh in the jsdoc?

Reply:
moved it to a new function and called in initDkg instead of the constructor which does feel more appropriate